backend/app/utils.py

```diff
@@ -0,0 +1,32 @@
+from yarl import URL
+
+
+def make_dsi(dataspace_base_domain: str, definition_path: str, source: str) -> str:
+    """
+    Construct a Data Source Identifier from the information on the request and token
+
+    DSIs are URIs in the format: dpp://<source>@<dataspace_base_domain>/<data_definition>
+    """
+
+    # Data sources are identified on dataspaces as <group> or <group>:<variant>, which on the DSI URI correspond to a
+    # "user" and an optional "password"
+    user, _, password = source.partition(":")  # Extract group:source to URL properties
+
+    if not password:
+        # Passing the value None will not append it to the URL, whereas an empty string would be
+        password = None
+
+    # Use a URL builder to build the URI correctly
+    #
+    # Our expectation is that all the values are suitable for doing this with just string concatenation, however for
+    # future proofing we want to do this a bit more carefully here.
+
+    dsi_url = URL.build(
+        scheme="dpp",
+        host=dataspace_base_domain,
```

Comment:
Do we actually need to pass in the `dataspace_base_domain` as a paremeter to this function, or could we rather just here read it from the config? This would make the calls to this a bit shorter. 

Of course we then lose ability to request DSIs for different dataspaces. On the other hand the repo is in all other ways structured to work with one dataspace at a time; we configure only one `DATASPACE_BASE_DOMAIN` and use it to fetch configs etc, so I can't see much point in making this function more versatile than the rest of the codebase when making it less versatile would make the calls to it a fair bit simpler.

Reply:
I guess this might be hard if we want to keep it in sync with the version in the example-productizer repo.

Reply:
I'll leave it up to you to consider if you want to keep it as it isor if you want to make the functions across the repos be out of sync and easier to use in both cases.

Reply:
I do want to keep it in sync